ui/pages/confirmations/components/confirm/info/approve/edit-spending-cap-modal/edit-spending-cap-modal.tsx

```diff
@@ -0,0 +1,177 @@
+    transactionMeta.txParams.to,
+    transactionMeta.txParams.from,
+    transactionMeta.txParams.data,
+  );
+
+  const accountBalance = getAccountBalance(userBalance || '0', decimals || '0');
+
+  const { formattedSpendingCap } = useApproveTokenSimulation(
+    transactionMeta,
+    decimals || '0',
+  );
+
+  const [customSpendingCapInputValue, setCustomSpendingCapInputValue] =
+    useState('');
+  const [customSpendingCap, setCustomSpendingCap] = useState('');
+  const setPositiveSpendingCap = (newValue: string) => {
+    const parsedValue = parseInt(newValue, 10);
+    // coerce negative numbers to zero
+    setCustomSpendingCap(parsedValue < 0 ? '0' : newValue);
+  };
+
+  const customTxParamsData = useMemo(() => {
+    return getCustomTxParamsData(transactionMeta?.txParams?.data, {
+      customPermissionAmount: customSpendingCap || '0',
+      decimals: decimals || '0',
+    });
+  }, [customSpendingCap, transactionMeta?.txParams?.data, decimals]);
+
+  const { estimatedGasLimit } = useTransactionGasEstimate(
+    transactionMeta,
+    customTxParamsData,
+    customSpendingCap,
+  );
+
+  useEffect(() => {
+    if (customSpendingCap && estimatedGasLimit) {
+      dispatch(
+        updateEditableParams(transactionMeta.id, {
+          data: customTxParamsData,
+          gas: hexToDecimal(estimatedGasLimit as string),
+        }),
+      );
+    }
+  }, [
+    customSpendingCap,
+    estimatedGasLimit,
+    customTxParamsData,
+    transactionMeta,
+    dispatch,
+  ]);
+
+  const handleCancel = () => {
+    setIsOpenEditSpendingCapModal(false);
+    setCustomSpendingCapInputValue('');
+  };
+
+  function handleSubmit() {
+    setIsOpenEditSpendingCapModal(false);
+    setCustomSpendingCapInputValue('');
+    setPositiveSpendingCap(customSpendingCapInputValue);
+  }
+
+  return (
+    <Modal
+      isOpen={isOpenEditSpendingCapModal}
+      onClose={() => setIsOpenEditSpendingCapModal(false)}
+      isClosedOnEscapeKey
+      isClosedOnOutsideClick
+      className="edit-spending-cap-modal"
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader
+          justifyContent={JustifyContent.center}
+          childrenWrapperProps={{
+            alignItems: AlignItems.center,
+            display: Display.Flex,
+            flexDirection: FlexDirection.Column,
+          }}
+        >
+          <Text variant={TextVariant.headingMd}>{t('editSpendingCap')}</Text>
+        </ModalHeader>
+        <ModalBody>
+          <Text
+            variant={TextVariant.bodyMd}
+            color={TextColor.textAlternative}
+            paddingBottom={4}
+          >
+            {t('editSpendingCapDesc')}
+          </Text>
+          <TextField
+            type={TextFieldType.Number}
+            value={customSpendingCapInputValue}
+            onChange={(event) =>
+              setCustomSpendingCapInputValue(event.target.value)
+            }
+            placeholder={`${
+              customSpendingCap === ''
```

Comment:
Will `customSpendingCap` always be empty as it's only set on submit which would close the modal?

Reply:
Again, submit closes the modal via the state hook but it doesn't unmount the component nor does it wipe other states.

Reply:
But even if it's not unmounting, does it not make the lifecycle simpler if we start from the same place each time?

Why do we need to persist anything from the last usage?

Can we not always use the formatted value?

Reply:
@matthewwalsh0 That would make this line simpler, but I don't think it's the expected solution from a user's perspective. Let's consider this scenario:

The dApp suggested spending cap is 3 DAI for example, and the user edits the spending cap to 5 DAI using the modal. He then opens the modal again to edit the spending cap to 6 DAI.

The first time the modal is opened we want to show 3 DAI as a placeholder, while the second time we want to show 5 DAI, not 3.

Reply:
Exactly, but we've updated the transaction in the controller state, so won't the `formattedSpendingCap` automatically reflect the latest value as it's parsed from the transaction metadata which is now different?

Reply:
I see your point now, and I agree. Pushed the changes 👍